src/Nethermind/Nethermind.GitBook/RpcAndCliGenerator.cs

```diff
@@ -0,0 +1,293 @@
+                    rpcBuilder.Append(returnBuilder);
+                    rpcBuilder.AppendLine();
+
+                    _markdownGenerator.OpenTabs(rpcBuilder);
+                    _markdownGenerator.CreateTab(rpcBuilder, $"Example request of {rpcMethodName}");
+                    _markdownGenerator.CreateCurlExample(rpcBuilder, rpcMethodName, exampleArguments);
+                    _markdownGenerator.CloseTab(rpcBuilder);
+
+                    if (methodData.ExampleResponse != null)
+                    {
+                        string exampleResponse = CreateRpcExampleResponse(methodData.ExampleResponse);
+                        _markdownGenerator.CreateTab(rpcBuilder, $"Example response of {rpcMethodName}");
+                        _markdownGenerator.CreateCodeBlock(rpcBuilder, exampleResponse);
+                        _markdownGenerator.CloseTab(rpcBuilder);
+                    }
+                    
+                    if (typesToDescribe.Count != 0)
+                    {
+                        objectsBuilder.AppendLine();
+                        _markdownGenerator.CreateTab(objectsBuilder, $"Objects in {rpcMethodName}");
+                        _sharedContent.AddObjectsDescription(objectsBuilder, typesToDescribe);
+                        _markdownGenerator.CloseTab(objectsBuilder);
+                    }
+
+                    rpcBuilder.Append(objectsBuilder);
+                    _markdownGenerator.CloseTabs(rpcBuilder);
+                    rpcBuilder.AppendLine();
+                    
+                    if (methodData.InvocationType == InvocationType.Both)
+                    {
+                        rpcBuilder.AppendLine($"[See also CLI {cliMethodName}](https://docs.nethermind.io/nethermind/nethermind-utilities/cli/{moduleName}#{moduleName.ToLower()}-{methodName.ToLower()})");
+                        
+                        CreateCliContent(cliBuilder, paramBuilder, returnBuilder, objectsBuilder, cliMethodName, exampleArguments, cliInvocation, methodData);
+                        cliBuilder.AppendLine($"[See also JSON RPC {rpcMethodName}](https://docs.nethermind.io/nethermind/ethereum-client/json-rpc/{moduleName}#{rpcMethodName.ToLower()})");
+                    }
+                }
+                else
+                {
+                    Type returnType = methodData.ReturnType;
+                    string returnTypeToWrite = _sharedContent.GetTypeToWrite(returnType, typesToDescribe);
+                    returnBuilder.AppendLine("| Returned type | Description |");
+                    returnBuilder.AppendLine("| :--- | :--- |");
+                    returnBuilder.AppendLine(@$"| `{returnTypeToWrite}` | {methodData.ResponseDescription} |");
+
+                    CreateCliContent(cliBuilder, paramBuilder, returnBuilder, objectsBuilder, cliMethodName, exampleArguments, cliInvocation, methodData);
+                }
+            }
+
+            const int notEmptyModule = 20;
```

Comment:
would be great to add a comment explaining where this 20 comes from or even do sth like:
const int emptyModuleLength = "this is empty module".Length;